src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx

```diff
@@ -67,6 +67,16 @@ const CreateTokenTransfer = ({
     }
   }, [setNonce, txNonce])
 
+  const balancesItems = useMemo(() => {
+    return isOnlySpendingLimitBeneficiary
+      ? balances.items.filter(({ tokenInfo }) => {
+          return spendingLimits?.some(({ beneficiary, token }) => {
+            return sameAddress(beneficiary, wallet?.address || '') && sameAddress(tokenInfo.address, token.address)
+          })
+        })
+      : balances.items
+  }, [balances.items, isOnlySpendingLimitBeneficiary, spendingLimits, wallet?.address])
```

Comment:
This is the same as it was before just pushed further up so we can use it to set the default form value.